### DISCUSSION TITLE
Genereated openapi.yaml, which meets swaggerhub standards

### DISCUSSION THREAD

**Discussion by Angel0926** [quarkusio/quarkus#32377] [Q&A]
Hi. can you answer me what i can add in application.yaml config file, so that meet swaggerhub standarts?   

i want add something in application.yaml, but i try such  alternative option-annotation @SecurityRequirement , in this case generate like this 
security:
- bearerAuth: []  (security and beareAuth in one line) but i want in other line 
<img width="238" alt="Screenshot 2023-04-03 at 22 07 40" src="https://user-images.githubusercontent.com/91777306/229591522-f4e6237b-e942-43c7-a9ac-d76c6ba3ad9d.png">

[a comment by an automated account is not shown]

**Comment by phillip-kruger** [comment]
Maybe this will help: https://quarkus.io/guides/security-overview-concept#authentication-mechanisms

**Reply by Angel0926** [reply]
no it's not. I ask about openapi.yaml generating process, and i want add some configs in application.yaml

**Reply by phillip-kruger** [reply]
openapi generated from the security model in the app. If you want to add some things manually, then you need to look at the MicroProfile OpenAPI annotations to add.  https://download.eclipse.org/microprofile/microprofile-open-api-3.1/microprofile-openapi-spec-3.1.html

**Reply by phillip-kruger** [reply]
You can also add things using a Filter. See https://download.eclipse.org/microprofile/microprofile-open-api-3.1/microprofile-openapi-spec-3.1.html#_oasfilter

**Comment by phillip-kruger** [comment]
But you probably want to use the security annotations: https://download.eclipse.org/microprofile/microprofile-open-api-3.1/microprofile-openapi-spec-3.1.html#_quick_overview_of_annotations

**Reply by Angel0926** [reply]
when i use [@SecurityRequirement] annotation in my controllers, the openapi.yaml generated good but not like this 
<img width="492" alt="Screenshot 2023-04-04 at 11 34 24" src="https://user-images.githubusercontent.com/91777306/229720906-4267a485-8e5b-4491-badc-a20b9b3e359a.png">
  and so 
<img width="264" alt="Screenshot 2023-04-04 at 11 50 20" src="https://user-images.githubusercontent.com/91777306/229725012-dd98a5f1-7f3c-4f0c-9fab-1be33698ffe0.png">
i don't want why its generate security and bearer auth in one line

**Reply by phillip-kruger** [reply]
You are going to need to define the security model as well

**Reply by Angel0926** [reply]
how?

**Reply by phillip-kruger** [reply]
see https://github.com/eclipse/microprofile-open-api/blob/master/api/src/main/java/org/eclipse/microprofile/openapi/annotations/security/SecurityScheme.java and https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.0.md#security-scheme-object

**Reply by Angel0926** [reply]
it is not help, i want know why it generate wrong and in one line

**Reply by phillip-kruger** [reply]
I can really say without seeing a reproducer, but it seems like there is not enough info for it to do what you want. What do you expect it to show ? With what input ?

**Reply by MikeEdgar** [reply]
@Angel0926 , is your concern with the YAML formatting of the `security` property in the OpenAPI?

You've stated that the actual output is like below, but what is it you expect to see?
```yaml
security:
  - bearerAuth: []
```

**Reply by Angel0926** [reply]
yes it is. actual output is not like this
 ```
 security:
   - bearerAuth: []
```
but like this 
```
  security:
  - bearerAuth: []
  
```
in same level

**Reply by MikeEdgar** [reply]
Those are equivalent. Are you saying some requirement somewhere for the additional indentation?

**Reply by Angel0926** [reply]
Yes for me it's equivalent too, but in swaggerhub  because of the indent i have a standartization problem.For example i send you screen from swaggerhub account

<img width="558" alt="Screenshot 2023-04-04 at 16 30 05" src="https://user-images.githubusercontent.com/91777306/229791893-feb1ba6d-6880-45be-8f23-f2d5c1f2521e.png">

<img width="438" alt="Screenshot 2023-04-04 at 16 30 19" src="https://user-images.githubusercontent.com/91777306/229791901-f718980e-801a-4704-8e7a-8c4424c15bff.png">

**Reply by MikeEdgar** [reply]
You need a `securitySchemes` entry named `bearerAuth` to address that error. Do you have anything defined in `components/securitySchemes` ?

**Reply by Angel0926** [reply]
only this in application.yaml 
```
quarkus:
  smallrye-openapi:
    security:
      -bearerAuth:
          type: http
          scheme: bearer
          bearerFormat: JWT  
    store-schema-directory: target/generated
```

and i added @SecurityRequirement(name = "bearerAuth", scopes = "Admin")  on controllers

**Reply by MikeEdgar** [reply]
Try this:
```yaml
quarkus:
  smallrye-openapi:
    security-scheme: jwt
    security-scheme-name: bearerAuth
```

**Comment by Angel0926** [comment]
and add @SecurityRequirement(name = "bearerAuth", scopes = "Admin") this annotation?

**Reply by MikeEdgar** [reply]
Yes, you can add that to the endpoints that require it. Note that `scopes` is only valid for OAuth2 and OIDC [1]. 

[1] https://spec.openapis.org/oas/v3.0.3.html#patterned-fields-2

**Reply by MikeEdgar** [reply]
What is the error? Maybe something to do with the scheme type of `http`.

**Comment by Angel0926** [comment]
everithing is ok) thanks a lot
